hr_job_category/models/hr.py

```diff
@@ -23,16 +23,32 @@ class HRJob(models.Model):
 class HRContract(models.Model):
     _inherit = "hr.contract"
 
-    def _tag_employees(self, job_id):
-        if job_id:
-            job = self.env["hr.job"].browse(job_id)
-            self.mapped("employee_id").write(
-                {"category_ids": [(6, 0, job.category_ids.ids)]}
-            )
-        else:
-            for contract in self:
-                categories = contract.job_id and contract.job_id.category_ids.ids or []
-                contract.employee_id.write({"category_ids": [(6, 0, categories)]})
+    def _remove_tags(self, employee_id=None, job_id=None):
+        # TODO write tags only once
+        if not employee_id or not job_id:
+            return
+        employee = self.env["hr.employee"].browse(employee_id)
+        empl_tags = employee.category_ids
+        job = self.env["hr.job"].browse(job_id)
+        _logger.debug(
+            "Removing employee tags if tag exists on contract " "job: %s", empl_tags
+        )
+        for tag in job.category_ids:
+            if tag in empl_tags:
+                employee.write({"category_ids": [(3, tag.id)]})
+
+    def _tag_employees(self, employee_id=None, job_id=None):
+        if not employee_id or not job_id:
+            return
+        employee = self.env["hr.employee"].browse(employee_id)
+        empl_tags = employee.category_ids
+        job = self.env["hr.job"].browse(job_id)
+        for tag in job.category_ids:
+            if tag not in empl_tags:
+                _logger.debug(
+                    "Adding employee tag if job tag doesn't " "exists: %s", tag.name
+                )
+                employee.write({"category_ids": [(4, tag.id)]})
```

Comment:
I would recommend to keep same method signature as before
```suggestion
    def _tag_employees(self, job_id):
        job = self.env["hr.job"].browse(job_id)
        _logger.debug(
            "Adding employee tags if job tags doesn't exist: %s", job.category_ids
        )
        for employee in self.mapped("employee_id"):
            tags_to_add = [
                (4, tag.id)
                for tag in job.category_ids - employee.category_ids
            ]
            if tags_to_add:
                employee.write(tags_to_add)
```